tools/topology/topology2/include/components/pipeline.conf

```diff
@@ -95,6 +95,20 @@ Class.Widget."pipeline" {
 		}
 	}
 
+	#
+	# be_pipeline - boolean indicating that it is a backend pipeline
```

Comment:
what is the definition of a backend pipeline?

This could be understood in two different ways
a) a pipeline that contains a dai_in or dai_out object
b) a pipeline that does not contain a PCM object or host copier

This needs to be clarified since we do need to support hierarchical mixing and topologies with 'intermediate' pipelines that are neither 'front-end' nor 'back-end'. I also don't know what front- and back-end mean for simple passthrough pipelines typically used initially for hardware validation.

Reply:
so why do you need a token if this can be determined by the driver by walking through the list of widgets for that pipelines?

I get that you need a boolean variable to decide to skip, but that information about the backend nature of the pipeline seems redundant.

Reply:
take the mixout-gain-dai-copier-playback.conf for example. When walking the list, if we encounter the mixout/gain widget first, how will we know if it is part of a backend pipeline or not without this token?

Reply:
you would need to walk through all elements of a pipeline to decide whether or not this flag needs to be set. This cannot be a per-widget decision, there's a first-pass needed.

Reply:
I don't understand that comment @plbossart . Why do we need a first pass? We already know If it's a backend pipe after parsing the token